File: sample_problematic_code.js
// Problematic JavaScript code for testing code review tools
// This file intentionally contains various syntax errors and anti-patterns

import {something,another,third} from 'module'; // Bad: no spaces after commas

// Magic numbers and poor function naming
function CalculateScore(data){ // Bad: no space before brace, CamelCase
var total=0; // Bad: var instead of let/const, no spaces
for(var i=0;i<data.length;i++){ // Bad: no spaces, var instead of let
if(data[i]>100){ // Bad: no spaces, magic number
total+=data[i]*2; // Bad: magic number
}else if(data[i]>50&&data[i]<=100){ // Bad: no spaces, magic numbers
total+=data[i]*1.5;
}else if(data[i]>25){
total+=data[i];
}else if(data[i]>10){
total+=data[i]*0.5;
}else{
total+=0;
}
}
return total;
}

// Class with too many responsibilities (God object)
class DataManager{ // Manager pattern + anti-pattern
constructor(){
this.data=[];
this.cache={};
this.settings={};
this.users=[];
this.logs=[];
this.stats={};
this.tempData=[];
this.processedItems=[];
this.failedItems=[];
this.successCount=0;
this.errorCount=0;
this.lastUpdated=null;
this.isDirty=false; // Too many properties
}

processData(input,mode,options,filters,validators,transformers,handlers,callbacks){ // Too many parameters
if(!input)return[]; // Bad: no space after if

let result=[];
let errors=[];

for(let i=0;i<input.length;i++){
let item=input[i];

if(mode==='strict'){
if(options&&options.validate){
if(validators){
for(let j=0;j<validators.length;j++){
let validator=validators[j];
if(validator==='email'){
if(typeof item!=='string'||item.indexOf('@')===-1||item.indexOf('.')===-1){
errors.push(`Invalid email at index ${i}`);
continue;
}
}else if(validator==='phone'){
if(typeof item!=='string'||item.length!==10){ // Magic number
errors.push(`Invalid phone at index ${i}`);
continue;
}
}else if(validator==='age'){
let age=parseInt(item);
if(age<18||age>120){ // Magic numbers
errors.push(`Invalid age at index ${i}`);
continue;
}
}
}
}
}

if(transformers){
for(let k=0;k<transformers.length;k++){
let transformer=transformers[k];
if(transformer==='uppercase'){
item=item.toString().toUpperCase();
}else if(transformer==='lowercase'){
item=item.toString().toLowerCase();
}else if(transformer==='trim'){
item=item.toString().trim();
}
}
}

result.push(item);
}else if(mode==='lenient'){
try{
result.push(item);
}catch(e){
errors.push(`Error processing item at index ${i}`);
}
}
}

return{result:result,errors:errors}; // Should return [result, errors] for consistency
} // Very long function with high complexity
}

function createUserFactory(type){ // Factory pattern
if(type==='admin'){
return{type:'admin',permissions:['read','write','delete']};
}else if(type==='user'){
return{type:'user',permissions:['read']};
}else{
return{type:'guest',permissions:[]};
}
}

class ConfigBuilder{ // Builder pattern
constructor(){
this.config={};
}

buildDatabaseConfig(){
this.config.database={host:'localhost',port:5432}; // Magic number
return this;
}

buildCacheConfig(){
this.config.cache={ttl:3600,maxSize:1000}; // Magic numbers
return this;
}

build(){
return this.config;
}
}

// Singleton pattern
class Logger{
constructor(){
if(Logger.instance){
return Logger.instance;
}
Logger.instance=this;
}

static getInstance(){
if(!Logger.instance){
Logger.instance=new Logger();
}
return Logger.instance;
}
}

// Syntax errors
function brokenFunction() {
let x = 10
let y = 20; // Missing semicolon above
if(x > y {  // Missing closing parenthesis
return x + y;
}

// Very long line that exceeds reasonable limits and should be flagged by any decent style checker as a violation that needs to be addressed immediately
let longVariable = "This is an extremely long string that definitely exceeds any reasonable line length limit and should be broken up into multiple lines for better readability and maintainability";

function processEverything(data,mode,settings,cache,logs,users,permissions,filters,validators,transformers,handlers,callbacks,options,config){ // Too many parameters
// Empty function
}

// Main execution with problems
if(__name__==="main"){ // Bad: should be different check for JS
let data=[1,2,3,42,100,1000]; // Magic numbers, no spaces
let score=CalculateScore(data);
console.log("Score: "+score); // Should use template literals

let manager=new DataManager();
let result=manager.processData(data,'strict',null,null,null,null,null,null);
console.log("Processed: "+result.result.length+" items");
}

File: sample_problematic_code.py
# Problematic Python code for testing code review tools
# This file intentionally contains various code smells and anti-patterns

import os,sys,json,re,time # Bad: multiple imports on one line

# Magic numbers throughout the code
def CalculateScore(data):  # Bad: CamelCase function name, no docstring
    total=0 # Bad: no spaces around operator
    for item in data:
        if item>100: # Bad: no spaces around operator, magic number
            total+=item*2 # Bad: no spaces, magic number
        elif item>50 and item<=100: # Bad: magic numbers, complex condition
            total+=item*1.5 # Bad: magic number
        elif item>25:
            total+=item
        elif item>10:
            total+=item*0.5
        elif item>5:
            total+=item*0.25
        elif item>1:
            total+=item*0.1
        else:
            total+=0

    if total>1000: # Magic number
        return total*1.2
    elif total>500: # Magic number
        return total*1.1
    elif total>200: # Magic number
        return total*1.05
    else:
        return total

class DataManager: # Manager pattern (good to detect)
    def __init__(self):
        self.data=[]
        self.cache={}
        self.settings={}
        self.users=[]
        self.logs=[]
        self.stats={}
        self.temp_data=[]
        self.processed_items=[]
        self.failed_items=[]
        self.success_count=0
        self.error_count=0
        self.last_updated=None
        self.is_dirty=False
        self.lock_status=False # Too many instance variables (God object)

    def processData(self,input_data,mode,options,filters,validators,transformers): # Bad: CamelCase, too many parameters
        if not input_data:
            return []

        result=[]
        errors=[]

        for i,item in enumerate(input_data):
            if mode=='strict':
                if options and 'validate' in options:
                    if validators:
                        for validator in validators:
                            if validator=='email':
                                if '@' not in str(item) or '.' not in str(item):
                                    errors.append(f"Invalid email at index {i}")
                                    continue
                            elif validator=='phone':
                                if len(str(item))!=10: # Magic number
                                    errors.append(f"Invalid phone at index {i}")
                                    continue
                            elif validator=='age':
                                if int(item)<18 or int(item)>120: # Magic numbers
                                    errors.append(f"Invalid age at index {i}")
                                    continue

                if transformers:
                    for transformer in transformers:
                        if transformer=='uppercase':
                            item=str(item).upper()
                        elif transformer=='lowercase':
                            item=str(item).lower()
                        elif transformer=='trim':
                            item=str(item).strip()

                if filters:
                    skip=False
                    for filter_rule in filters:
                        if filter_rule=='empty':
                            if not item or str(item).strip()=='':
                                skip=True
                                break
                        elif filter_rule=='numeric':
                            try:
                                float(item)
                            except:
                                skip=True
                                break
                    if skip:
                        continue

                result.append(item)

            elif mode=='lenient':
                try:
                    if transformers:
                        for transformer in transformers:
                            if transformer=='clean':
                                item=str(item).replace('  ',' ').strip()
                    result.append(item)
                except:
                    errors.append(f"Error processing item at index {i}")

        return result,errors # Long function with high complexity

def create_user_factory(user_type): # Factory pattern (good to detect)
    if user_type=='admin':
        return {'type':'admin','permissions':['read','write','delete']}
    elif user_type=='user':
        return {'type':'user','permissions':['read']}
    else:
        return {'type':'guest','permissions':[]}

class ConfigBuilder: # Builder pattern (good to detect)
    def __init__(self):
        self.config={}

    def build_database_config(self):
        self.config['database']={'host':'localhost','port':5432} # Magic number
        return self

    def build_cache_config(self):
        self.config['cache']={'ttl':3600,'max_size':1000} # Magic numbers
        return self

    def build(self):
        return self.config

# Singleton pattern (good to detect)
class Logger:
    _instance=None

    def __new__(cls):
        if cls._instance is None:
            cls._instance=super().__new__(cls)
        return cls._instance

# Very long line that exceeds the recommended 79 character limit and should be flagged by the style checker as a violation
x = "This is an extremely long string that goes way beyond the recommended line length limit and should definitely be broken into multiple lines for better readability and maintainability"

def process_everything(data,mode,settings,cache,logs,users,permissions,filters,validators,transformers,handlers): # Too many parameters
    pass # Empty function

if(__name__=="__main__"): # Bad: missing space, should be if __name__ == "__main__":
    data=[1,2,3,42,100,1000] # Magic numbers
    score=CalculateScore(data)
    print(f"Score: {score}")

    manager=DataManager()
    result,errors=manager.processData(data,'strict',None,None,None,None)
    print(f"Processed: {len(result)} items")